app/services/importers/okc_blotter/pdf.rb

```diff
@@ -0,0 +1,139 @@
+require 'zip'
+
+module Importers
+  module OkcBlotter
+    class Pdf < ApplicationService
+      def initialize(date)
+        @date = date
+      end
+
+      def perform
+        download_pdf_from_website unless pdf
+        save
+      end
+
+      def download_pdf_from_website
+        input = HTTParty.get("#{url}/pdfs?date=#{@date}",
+                             headers: {
+                               Authorization: auth_token
+                             }).body
+        Zip::InputStream.open(StringIO.new(input)) do |io|
+          while entry = io.get_next_entry #note: there is only actually one pdf here
+            pdf = io.read
+            filename = entry.name
+            Bucket.new.put_object("#{s3_path}/#{filename}", pdf)
+          end
+        end
+      end
+
+      def save
+        bookings_hash = json
+        ::OkcBlotter::Pdf.create!(
+          parsed_on: DateTime.now,
+          date: @date,
+          bookings: bookings_hash.map do |booking_hash|
+            build_bookings(booking_hash)
+          end
+        )
+      end
+
+      def json
+        def pdf.path
+          '/path/hack.pdf' # this is a hack see https://github.com/jnunemaker/httparty/issues/675#issuecomment-590757288
+        end
+
+        return @json if @json
+
+        response = HTTParty.post(
+          "#{url}/parse",
+          body: { pdf: pdf },
+          headers: { Authorization: auth_token }
+        )
+        @json = JSON.parse(response.body)
+        return @json if response.success?
+
+        raise StandardError.new "Error parsing pdf: #{@json['message']}"
+      end
+
+      def pdf
+        return @pdf if @pdf
+        pdf = Bucket.new.get_object("#{s3_path}/#{@date}.pdf").body rescue false
```

Comment:
where pdf is defined, if null runs the download_pdf_from_website method. pdf found in s3 bucket/folder that matches corresponding date